# Airbnb Amsterdam Data Analysis

## Project Overview

This notebook  explores the Airbnb Amsterdam dataset using Python and Pandas.

The goals of this analysis are to:

1. Assess the quality of the dataset.
2. Clean the data where necessary.
3. Explore relationships between variables.
4. Answer business research questions using data analysis.

### Research Questions

1. Which neighbourhood has the highest Airbnb Listing prices?
2. How does room type affect Airbnb Listing prices?
3. Are there price outliers,and how do they affect the analysis?

In [1]:
## Import Libraries

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from ydata_profiling import ProfileReport

/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm
/var/folders/7r/z88gd3sd0rv_x899l45rkmm40000gn/T/ipykernel_18259/1422126751.py:4: DeprecationWarning: 
    `import ydata_profiling` is deprecated and will not receive more updates. 
    Please use `import data_profiling` instead.
    
  from ydata_profiling import ProfileReport


In [3]:
## Laod Dataset

In [4]:
df = pd.read_csv("../data/airbnb.csv")
df.head()

,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365,number_of_reviews_ltm,license
0,27886,"Romantic, stylish B&B houseboat in canal district",97647,Flip,NaN,Centrum-West,52.387610,4.89188,Private room,132.0,3,311,2025-09-07,1.87,1,17,33,0363 974D 4986 7411 88D8
1,28871,Comfortable double room,124245,Edwin,NaN,Centrum-West,52.367750,4.89092,Private room,89.0,2,732,2025-09-07,3.99,2,126,93,0363 607B EA74 0BD8 2F6F
2,29051,Comfortable single / double room,124245,Edwin,NaN,Centrum-Oost,52.365840,4.89111,Private room,61.0,2,849,2025-09-08,4.81,2,95,86,0363 607B EA74 0BD8 2F6F
3,44391,Quiet 2-bedroom Amsterdam city centre apartment,194779,Jan,NaN,Centrum-Oost,52.371680,4.91471,Entire home/apt,NaN,3,42,2022-08-20,0.23,1,0,0,0363 E76E F06A C1DD 172C
4,48373,Cozy family home in Amsterdam South,220434,Vesna & Misha,NaN,Buitenveldert - Zuidas,52.327808,4.87680,Entire home/apt,NaN,3,5,2024-04-28,0.19,1,0,0,0363 4A2B A6AD 0196 F684


In [5]:
## Initial Data Exploration

df.shape

df.info()

df.describe()

df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10480 entries, 0 to 10479
Data columns (total 18 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              10480 non-null  int64  
 1   name                            10480 non-null  object 
 2   host_id                         10480 non-null  int64  
 3   host_name                       10477 non-null  object 
 4   neighbourhood_group             0 non-null      float64
 5   neighbourhood                   10480 non-null  object 
 6   latitude                        10480 non-null  float64
 7   longitude                       10480 non-null  float64
 8   room_type                       10480 non-null  object 
 9   price                           5874 non-null   float64
 10  minimum_nights                  10480 non-null  int64  
 11  number_of_reviews               10480 non-null  int64  
 12  last_review                     

id                                    0
name                                  0
host_id                               0
host_name                             3
neighbourhood_group               10480
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                              4606
minimum_nights                        0
number_of_reviews                     0
last_review                        1097
reviews_per_month                  1097
calculated_host_listings_count        0
availability_365                      0
number_of_reviews_ltm                 0
license                             119
dtype: int64

In [6]:
## Data Quality Assessment

In [7]:
profile = ProfileReport (
    df, title = "Airbnb Amsterdam Data Quality Report",
    explorative = True
)

In [8]:
profile.to_file("../reports/airbnb_report.html")

Summarize dataset:   9%| | 2/23 [00:00<00:00, 24.83
Summarize dataset:  48%|▍| 11/23 [00:00<00:00, 27.7
100%|██████████████| 18/18 [00:00<00:00, 94.78it/s]
Summarize dataset: 100%|█| 149/149 [00:06<00:00, 24
Generate report structure: 100%|█| 1/1 [00:02<00:00
Render HTML: 100%|███| 1/1 [00:00<00:00,  2.25it/s]
Export report to file: 100%|█| 1/1 [00:00<00:00, 89


### Overview Observations

The Airbnb Amsterdam dataset contains **10,480 observations and **18 variables**.Data profiling identified **17,402 missing values (9.2%)**,indicating that data cleaning will be necessary before performing further analysis.

No duplicate records were found, which suggests the dataset is free from repeated listings. The dataset occupies approximately **4.9 MiB** of memory and contains a mixture of numeric, categorical, text, and datetime variables suitable for exploratory data analysis.

In [9]:
df[df["price"].isna()].head(10)

,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365,number_of_reviews_ltm,license
3,44391,Quiet 2-bedroom Amsterdam city centre apartment,194779,Jan,NaN,Centrum-Oost,52.371680,4.914710,Entire home/apt,NaN,3,42,2022-08-20,0.23,1,0,0,0363 E76E F06A C1DD 172C
4,48373,Cozy family home in Amsterdam South,220434,Vesna & Misha,NaN,Buitenveldert - Zuidas,52.327808,4.876800,Entire home/apt,NaN,3,5,2024-04-28,0.19,1,0,0,0363 4A2B A6AD 0196 F684
9,53921,Amsterdam Stylish Lakeview Apartment,252245,Ingrid,NaN,IJburg - Zeeburgereiland,52.355590,5.003200,Entire home/apt,NaN,1,12,2024-05-25,0.14,1,0,0,0363 B43C B1D4 2666 3739
10,55709,Bright Loft in Centre Amsterdam,263233,Jan-Willem,NaN,Centrum-Oost,52.358112,4.897637,Entire home/apt,NaN,11,69,2025-05-13,0.38,1,0,1,0363 E295 7D15 8420 343E
11,55868,Apartment near Museumplein (centre),264178,Cornelie,NaN,Zuid,52.355970,4.885120,Entire home/apt,NaN,3,96,2023-08-07,0.54,1,0,0,0363 B5A3 5F63 B032 A320
12,62015,"Charming, beautifully & sunny place",301340,Jessica,NaN,Oud-Oost,52.364310,4.931220,Entire home/apt,NaN,30,38,2024-03-31,0.21,1,22,0,0363 B090 499C D757 7F4F
13,80635,TOP LOCATED Canalhouse B&B Jordaan,436145,Riks,NaN,Centrum-West,52.378280,4.894400,Private room,NaN,3,241,2022-12-31,1.37,4,0,0,0363 4108 6EC3 C2FF 2493
14,91535,"Privat room and shared Houseboat ""Ramona"" centre",494281,Michiel,NaN,De Pijp - Rivierenbuurt,52.357740,4.896470,Private room,NaN,1,318,2025-09-07,1.85,1,0,71,0363 74BA 2074 E0CF 1CB8
17,155548,Apartment 2 minutes from Centrum!,748161,Maria,NaN,Oud-Oost,52.364200,4.928290,Entire home/apt,NaN,3,21,2019-08-20,0.12,1,0,0,0363 A562 4D7E 4A91 60DF
18,158633,"Typical Dutch house, quiet area, close to centre",761321,Louisa,NaN,Noord-Oost,52.393640,4.945380,Entire home/apt,NaN,3,20,2025-08-30,0.12,1,0,3,0363 D141 1024 C60E 96BA


In [10]:
df_clean = df.dropna(subset=["price"])

In [11]:
df_clean.shape

(5874, 18)

In [12]:
df_clean = df_clean.drop(columns=["neighbourhood_group"])

In [13]:
df_clean.shape

(5874, 17)

## Data Quality Report (Cleaned Dataset)

In [14]:
profile_clean = ProfileReport(
    df_clean,
    title="Cleaned Airbnb Amsterdam Data Quality Report",
    explorative=True
)

In [15]:
profile_clean.to_file("../reports/airbnb_clean_report.html")

Summarize dataset:   9%| | 2/22 [00:00<00:00, 23.50
Summarize dataset:  73%|▋| 16/22 [00:00<00:00, 83.5
100%|█████████████| 17/17 [00:00<00:00, 150.64it/s]
Summarize dataset: 100%|█| 148/148 [00:05<00:00, 24
Generate report structure: 100%|█| 1/1 [00:02<00:00
Render HTML: 100%|███| 1/1 [00:00<00:00,  3.48it/s]
Export report to file: 100%|█| 1/1 [00:00<00:00, 10
